lib/excon/middlewares/idempotent.rb

```diff
@@ -20,6 +20,9 @@ def error_call(datum)
 
         if datum[:idempotent] && [Excon::Errors::Timeout, Excon::Errors::SocketError,
             Excon::Errors::HTTPStatusError].any? {|ex| datum[:error].kind_of?(ex) } && datum[:retries_remaining] > 1
+
+          sleep(datum[:retry_interval]) if datum[:retry_interval] && !datum[:mock]
```

Comment:
I wonder if we might want this to occur even in mock mode? If you want to disable it then, you could change the interval, but I think just skipping it might be surprising in some cases. That said, any testing that relates to timing can be rather brittle.